cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -1173,5 +1175,173 @@ TEST(Substrait, JoinPlanInvalidKeys) {
           &ext_set));
 }
 
+TEST(Substrait, SerializeRelation) {
+#ifdef _WIN32
+  GTEST_SKIP() << "ARROW-16392: Substrait File URI not supported for Windows";
+#else
+  ExtensionSet ext_set;
+  auto dummy_schema = schema({field("foo", binary())});
+  // creating a dummy dataset using a dummy table
+  auto format = std::make_shared<arrow::dataset::ParquetFileFormat>();
+  auto filesystem = std::make_shared<fs::LocalFileSystem>();
```

Comment:
I think we can use a mock filesystem here instead of relying on PARQUET_TEST_DATA.  We aren't actually reading the file so it doesn't need any contents.

Reply:
I think there was an issue using Mock filesystem as I can recall with Substrait related things. In a previous PR for Substrait we also used as we have done here. I cannot excatly recall why we didn't use Mock filesystem, but I can try.

Reply:
WDYT?

Reply:
Please do try.  I would like to reduce our dependence on PARQUET_TEST_DATA as much as possible

Reply:
@westonpace I removed the `PARQUET_TEST_DATA` usage from the the current test case, but I had to add `parquet` dependency for the tests. Is it acceptable? Also the older tests are still using `PARQUET_TEST_DATA`, shall we replace it in this PR itself?

Reply:
On second thoughts. I will handle it later. It may need a little more generic setting. The default one doesn't work well for other tests. For now I am addressing it just for this test case.